recipes-extended/xen/files/libxl-openxt-tweaks.patch

```diff
@@ -0,0 +1,133 @@
+################################################################################
+SHORT DESCRIPTION:
+################################################################################
+LibXL modifications specific to OpenXT
+
+################################################################################
+LONG DESCRIPTION:
+################################################################################
+* Write a couple of xenstore nodes
+* Use a pty as the console backend
+* Fix domain unpause for disk-less Linux stubdomains
+
+################################################################################
+CHANGELOG
+################################################################################
+Authors:
+Martin Osterloh <osterlohm@ainfosec.com>
+
+################################################################################
+REMOVAL
+################################################################################
+Probably not desired.
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+This is specific to OpenXT, nothing to upstream here.
+
+################################################################################
+INTERNAL DEPENDENCIES
+################################################################################
+libxl-RFC-*.patch
+
+################################################################################
+PATCHES
+################################################################################
+Index: xen-4.6.1/tools/libxl/libxl_dm.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_dm.c
++++ xen-4.6.1/tools/libxl/libxl_dm.c
+@@ -1579,6 +1579,11 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+                                        libxl__xs_get_dompath(gc, guest_domid)),
+                         "%s",
+                         libxl_bios_type_to_string(LIBXL_BIOS_TYPE_SEABIOS));
++        /* OpenXT: We use legacy roms, which is disabled by default in sebios */
++        libxl__xs_write(gc, XBT_NULL,
++                        libxl__sprintf(gc, "%s/hvmloader/seabios-legacy-load-roms",
++                                       libxl__xs_get_dompath(gc, guest_domid)),
++                        "1");
+     }
+     ret = xc_domain_set_target(ctx->xch, dm_domid, guest_domid);
+     if (ret<0) {
+@@ -1606,6 +1611,12 @@ retry_transaction:
+         if (errno == EAGAIN)
+             goto retry_transaction;
+ 
++    /* OpenXT: We add the device models extended power management type to enable acpi.
++     * We do this here since we need the permission above.
++     */
++    libxl__xs_write(gc, XBT_NULL,
```

Comment:
I think we can get rid of this since we only support 1 mode now. I can make a ticket to clean it up later.